src/addrdb.cpp

```diff
@@ -109,15 +109,15 @@ template <typename Data>
 bool DeserializeFileDB(const fs::path& path, Data& data)
 {
     // open input file, and associate with CAutoFile
-    FILE *file = fsbridge::fopen(path, "rb");
+    FILE* file = fsbridge::fopen(path, "rb");
     CAutoFile filein(file, SER_DISK, CLIENT_VERSION);
-    if (filein.IsNull())
-        return error("%s: Failed to open file %s", __func__, path.string());
-
+    if (filein.IsNull()) {
+        LogPrintf("Missing or invalid file %s\n", path.string());
+        return false;
+    }
     return DeserializeDB(filein, data);
 }
```

Comment:
I think the word "invalid" implies that something is wrong with the contents of the file, maybe could be misleading to use it for a file open failure.

What about distinguishing a missing file from another open error and staying silent on missing file (since that is normal) and printing a screaming error otherwise? Also print the actual error that has occurred - the user wants to see if it is something like "Disk IO error".

```cpp
    if (!fs::exists(path)) {
        return false;
    }
    FILE* file = fsbridge::fopen(path, "rb");
    if (file == nullptr) {
        return error("%s: Failed to open file %s: %s", __func__, path.string(),
                     fsbridge::GetErrorReason()); // would require to make GetErrorReason() public
    }
    CAutoFile filein(file, SER_DISK, CLIENT_VERSION);
    return DeserializeDB(filein, data);
```

Reply:
Thanks, done in the last commit.